core/chain-configs/src/genesis_config.rs

```diff
@@ -571,6 +571,14 @@ impl Genesis {
         stream_records_from_file(reader, callback).map_err(io::Error::from)
     }
 
+    /// Returns number of records in the genesis or path to records file.
+    pub fn records_len(&self) -> Result<usize, &Path> {
```

Comment:
This looks like a rather adhoc API. I see that we need it to preserve the behavior exactly, including logging and such, but perhaps it'll be better to just adjust log-lines?

Reply:
I think this is best done in subsequent commit.  Specifically I’m thinking if we had `pub enum GenesisRecords { Records(Vec<_>), File(PathBuf) }` than places in the code which now use records_len would just match that enum.

Reply:
>Specifically I’m thinking if we had

One potential complication there is that we "need" the records field for JSON deserialization.

Reply:
This can be done by implementing serde::Serialize and
serde::Deserialize for the type explicitly or introducing GenesisJSON
struct which we convert to and from.  That’s a future implementer
problem. ;)